skglm/utils/prox_funcs.py

```diff
@@ -60,15 +60,26 @@ def value_MCP(w, alpha, gamma):
 
 
 @njit
-def prox_MCP(value, stepsize, alpha, gamma):
-    """Compute the proximal operator of stepsize * MCP penalty."""
+def value_weighted_MCP(w, alpha, gamma, weights):
+    """Compute the value of the weighted MCP."""
+    s0 = np.abs(w) < gamma * alpha
+    value = np.zeros_like(w)
+    value[~s0] = weights[~s0] * gamma * alpha ** 2 / 2.
+    value[s0] = alpha * np.abs(w[s0]) - w[s0]**2 / (2 * gamma)
+    value[s0] *= weights[s0]
+    return np.sum(value)
+
+
+@njit
+def prox_MCP(value, stepsize, alpha, gamma, positive=False, weight=1.):
```

Comment:
I'm not sure the maths are correct
With positive MCP we add $\delta_{x \geq0 }$, hence we should end up with [^1]<sup>(Eq 2 and below)</sup>

$$
\mathrm{prox} = \Pi_{\mathbb{R}} \circ \mathrm{prox}_{\mathrm{MCP}}
$$

Which I don't think reflects the last condition.
Any thoughts @mathurinm?

[^1]: Pustelnik, N. and Condat, L., 2017. Proximity operator of a sum of functions; application to depth map estimation. IEEE Signal Processing Letters, 24(12), pp.1827-1831.

Reply:
Why is not reflected in the last condition ? proxMCP has the same sign as its input

Reply:
That's right, thanks for the clarification!